storage/allocator_test.go

```diff
@@ -455,6 +457,109 @@ func TestAllocatorRebalance(t *testing.T) {
 	}
 }
 
+// TestAllocatorRebalanceThrashing tests that the rebalancer does not thrash
+// when replica counts are balanced, within the appropriate thresholds, across
+// stores.
+func TestAllocatorRebalanceThrashing(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	stopper, g, _, a, _ := createTestAllocator()
+	a.options.Deterministic = true
+	defer stopper.Stop()
+
+	stores := []*roachpb.StoreDescriptor{
```

Comment:
is the fact that there are 4 stores coupled to the fact that there are 4 ranges in the factory functions below? if not, better to make the numbers different to avoid chekhov's gun.